### PR TITLE
Axios master

### PR DESCRIPTION
**Pull request by osho-20**
https://browserstack.atlassian.net/browse/SDK-1671

recovered a11y and o11y code and change http call from request to axios

### CODE REVIEW

**Comment by osho-20** [comment]
@sauravdas1997 can you review once

File: test/unit/bin/helpers/reporterHTML.js [hunk 1]
```diff
@@ -233,27 +227,27 @@ describe('reporterHTML', () => {
...
-      pathStub.restore();
-    });
-
-    reporterHTML.reportGenerator(bsConfig, buildId, args, rawArgs, {});
-
-    sinon.assert.calledOnce(requestStub);
-    sinon.assert.calledOnce(getUserAgentStub);
-    sendUsageReportStub.calledOnceWithExactly(bsConfig, args, message, messageType, errorCode, {}, rawArgs);
-  });
+  // describe('generateCypressBuildReport', () => {
```
Comment (sauravdas1997) [inline test/unit/bin/helpers/reporterHTML.js:230]: why did we remove test ?

File: bin/testObservability/reporter/index.js [hunk 2]
```diff
@@ -0,0 +1,762 @@
...
+          debugOnConsole(`[MOCHA EVENT] EVENT_TEST_PENDING for uuid: ${test.testAnalyticsId}`);
+          if(!this.runStatusMarkedHash[test.testAnalyticsId]) {
+            this.runStatusMarkedHash[test.testAnalyticsId] = true;
+            await this.sendTestRunEvent(test,undefined,false,"TestRunSkipped");
+          }
+        }
+      })
+
+      .on(EVENT_TEST_BEGIN, async (test) => {
+        debugOnConsole(`[MOCHA EVENT] EVENT_TEST_BEGIN`);
```
Comment (sauravdas1997) [inline bin/testObservability/reporter/index.js:172]: this looks redundant

File: bin/testObservability/reporter/index.js [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+        debugOnConsole(`[MOCHA EVENT] EVENT_TEST_BEGIN for uuid: ${test.testAnalyticsId}`);
+        if (this.runStatusMarkedHash[test.testAnalyticsId]) return;
+        if(this.testObservability == true) {
+          await this.testStarted(test);
+        }
+      })
+
+      .on(EVENT_TEST_END, async (test) => {
+        debugOnConsole(`[MOCHA EVENT] EVENT_TEST_END`);
```
Comment (sauravdas1997) [inline bin/testObservability/reporter/index.js:181]: Redundant

File: bin/testObservability/reporter/index.js [hunk 4]
```diff
@@ -0,0 +1,762 @@
...
+        await this.uploadTestSteps();
+      });
+  }
+
+  registerListeners() {
+    startIPCServer(
+      (server) => {
+        server.on(IPC_EVENTS.CONFIG, this.cypressConfigListener.bind(this));
+        server.on(IPC_EVENTS.LOG, this.cypressLogListener.bind(this));
+        server.on(IPC_EVENTS.SCREENSHOT, this.cypressScreenshotListener.bind(this));
```
Comment (sauravdas1997) [inline bin/testObservability/reporter/index.js:216]: did we test screenshots in o11y ?

File: bin/testObservability/crashReporter/index.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+              }
+          })
+          .catch(error => debug(`[Crash_Report_Upload] Failed due to ${error}`));
+    } catch(e) {
+      debug(`[Crash_Report_Upload] Processing failed due to ${e && e.stack}`);
+    }
+  }
+}
+
+module.exports = CrashReporter;
```
Comment (sauravdas1997) [inline bin/testObservability/crashReporter/index.js:173]: EOF

File: bin/helpers/atsHelper.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    return {};
+  }
+}
+
+exports.atsFileCleanup = (bsConfig) => {
+  const filePath = path.resolve(bsConfig.run_settings.patched_cypress_config_file);
+  if(fs.existsSync(filePath)){
+    fs.unlinkSync(filePath);
+  }
+}
```
Comment (sauravdas1997) [inline bin/helpers/atsHelper.js:116]: EOF

File: bin/helpers/atsHelper.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
+const path = require('path');
```
Comment (sauravdas1997) [inline bin/helpers/atsHelper.js:1]: is ATS Tested ?

File: bin/commands/runs.js [hunk 8]
```diff
@@ -117,9 +145,36 @@ module.exports = function run(args, rawArgs) {
...
+          Constants.turboScaleObj.uploadUrl = gridDetails.cypressUrl + '/upload';
+          Constants.turboScaleObj.buildUrl = gridDetails.cypressUrl + '/build';
+
+          logger.debug(`Automate TurboScale Grid URL set to ${gridDetails.url}`);
+
+          patchCypressConfigFileContent(bsConfig);
+        } else {
+          process.exitCode = Constants.ERROR_EXIT_CODE;
+          return;
+        }
```
Comment (sauravdas1997) [inline bin/commands/runs.js:173]: These are ATS related handling, this needs to be verified by ATS team

File: bin/accessibility-automation/cypress/index.js [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+
+        cy.window().then(async (win) => {
+            await performScan(win);
+            browserStackLog('Getting accessibility results');
+            return await getAccessibilityResults(win);
+        });
+
+    } catch {}
+
+});
```
Comment (sauravdas1997) [inline bin/accessibility-automation/cypress/index.js:381]: EOF

**Review by sauravdas1997 — CHANGES_REQUESTED** [review]
- Handling of ATS changes are done by us, this needs to be revisited and tested by ATS team
- Verify the include/exclude tags feature for A11Y

File: bin/testObservability/reporter/index.js (hunk 4 above)
Comment (osho-20) [reply bin/testObservability/reporter/index.js:216]: yes

**Review by sauravdas1997 — APPROVED** [review]

**Pull request closed without merging**